File: Day-08-of-30-Days/Solution.java
// Day 08 of 30 Days 👾
public class Solution {
    public static boolean isPowerOfTwo(int num) {
        if (num < 1) {
            return false;
        }
        return (num & (num - 1)) == 0;
    }

    // Example
    public static void main(String[] args) {
        System.out.println(isPowerOfTwo(6)); // false
    }
}

File: Day-08-of-30-Days/Solution.c
// Day 08 of 30 Days 👾

#include <stdio.h>

    int
    powerOfTwo(int num) {
    if (num < 1) {
        return 0;
    }
    return (num & (num - 1)) == 0;
}

int main() {
    printf("%d\n", powerOfTwo(6)); // return 0 means false
    printf("%d\n", powerOfTwo(8)); // return 1 means true
    return 0;
}